genkan/auth/oAuth.js

```diff
@@ -0,0 +1,182 @@
+                        const hashedPasswordSHA512 = sha512({
+                            a: password,
+                            b: email + config.genkan.secretKey,
+                        })
+
+                        // Bcrypt Hashing
+                        const hashedPasswordSHA512Bcrypt = bcrypt.hashSync(hashedPasswordSHA512, saltRounds)
+
+                        // Generate email confirmation token
+                        const emailConfirmationToken = tokenGenerator()
+
+                        const NewUserSchema = {
+                            'email': email,
+                            'password': hashedPasswordSHA512Bcrypt,
+                            'googleID': googleID,
+                            'account': {
+                                'activity': {
+                                    'created': new Date(),
+                                    'lastSeen': null,
+                                },
+                                'type': 'STANDARD',
+                                'suspended': false,
+                                'emailVerified': false,
+                            },
+                            'tokens': {
+                                'emailConfirmation': emailConfirmationToken,
+                            },
+                        }
+
+                        // Insert new user into database
+                        insertDB(db, 'users', NewUserSchema, () => {
+                            findUIDByGoogleID(email, null, (result) => {
+                                // Schema for sessions in session collection
+                                const SessionSchema = {
+                                    'uid': result,
+                                    'sid': tokenGenerator(),
+                                    'timestamp': (new Date()).toISOString(),
+                                    'createdTimestamp': new Date(),
+                                }
+                                // Payload to update user's last seen in users collection
+                                const UpdateLastSeenPayload = {
+                                    $set: {
+                                        'account.activity.lastSeen': new Date(),
+                                    },
+                                }
+                                insertDB(db, 'sessions', SessionSchema, () => {
+                                    updateDB(db, 'users', {'email': email}, UpdateLastSeenPayload, () => {
+                                        sendConfirmationEmail(email, emailConfirmationToken)
+                                        return callback(sid)
+                                    })
+                                })
+                            })
+                        })
+                    }
+                })
+            }
+        })
+    }
+
+    // This is for the inserting uid into cookie
+    findUIDByGoogleID = (email, googleID, callback) => {
```

Comment:
In a usual cookie by Genkan (without OAuth), UID isn't inserted because the UID is queried using the SID.